fakesnow/conn.py

```diff
@@ -29,6 +30,7 @@ def __init__(
         create_schema: bool = True,
         db_path: str | os.PathLike | None = None,
         nop_regexes: list[str] | None = None,
+        sfqid_results: dict[str, pyarrow.Table] | None = None,
```

Comment:
should never be `None` - it needs to be created and owned outside so it can be shared across connections
```suggestion
results_cache: dict[str, pyarrow.Table],
```
also `results_cache` is what snowflake calls this